test/unit/specs/core/network/sendBeaconFactory.spec.js

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import sendBeaconFactory from "../../../../../src/core/network/sendBeaconFactory";
+
+describe("sendBeaconFactory", () => {
+  it("falls back to fetch if sendBeacon fails", () => {
+    const navigator = {
+      sendBeacon: jasmine.createSpy().and.returnValue(false)
+    };
+    const fetchPromise = Promise.resolve();
+    const fetch = jasmine.createSpy().and.returnValue(fetchPromise);
+    const logger = {
+      log: jasmine.createSpy()
+    };
+    const sendBeacon = sendBeaconFactory(navigator, fetch, logger);
+    const body = { a: "b" };
+    const result = sendBeacon("https://example.com/endpoint", body);
+    expect(navigator.sendBeacon).toHaveBeenCalledWith(
```

Comment:
It may be worth it to add a check for the context of the call.  When I was developing sendBeacon I wanted to just give it navigator.sendBeacon, but the context matters.
```
expect(navigator.sendBeacon.calls.first().object).toBe(navigator)
```